c7n/resources/xray.py

```diff
@@ -0,0 +1,105 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from c7n.actions import ActionRegistry, BaseAction
+from c7n.filters import FilterRegistry, Filter
+from c7n.manager import resources
+from c7n.query import QueryResourceManager
+from c7n.utils import local_session, type_schema
+
+
+filters = FilterRegistry('xray.filters')
+actions = ActionRegistry('xray.actions')
+
+
+@resources.register('xray')
+class Xray(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'xray'
+        id = name = 'Name'
+        dimension = None
+        filter_name = None
+        enum_spec = ('get_encryption_config', 'EncryptionConfig', None)
+
+    filter_registry = filters
+    action_registry = actions
+
+    def __init__(self, data, options):
+        super(Xray, self).__init__(data, options)
+
+
+@filters.register('encrypt-key')
+class XrayEncrypted(Filter):
+    """Determine if xray is encrypted.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: xray-encrypt-with-default
+                resource: xray
+                filters:
+                  - type: encrypt-key
+                    key: default
+              - name: xray-encrypt-with-kms
+                  - type: encrypt-key
+                    key: kms
+    """
+
+    permissions = ('xray:GetEncryptionConfig',)
+    schema = type_schema(
+        'encrypt-key',
+        required=['key'],
+        key={'type': 'string', 'enum': ['default', 'kms']}
+    )
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('xray')
+        return [{'key': client.get_encryption_config()['EncryptionConfig']['KeyId']}]
+
+
+@actions.register('encrypt')
+class SetXrayEncryption(BaseAction):
+    """Enable specific xray encryption.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: xray-default-encrypt
+                resource: xray
+                actions:
+                  - type: encrypt
+                    key: default
+              - name: xray-kms-encrypt
+                  - type: encrypt
+                    key: alias/some/alias/key
+    """
+
+    permissions = ('xray:PutEncryptionConfig',)
+    schema = type_schema(
+        'encrypt',
+        required=['key'],
+        key={'type': 'string'}
+    )
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('xray')
+        key = self.data.get('key')
+        req = {'Type': 'None'} if key == 'default' else {'Type': 'KMS', 'KeyId': key}
```

Comment:
Thats a invalid value 'None', the literal string is 'NONE'. also this should be a no-op if the value is already set to the target value instead of an api call.